### PR TITLE
Add KitClaimEvent

### LINKED ISSUE #3196: Add an Event for when a player uses a kit
Add an Event for when a player uses a kit

### PR DESCRIPTION
**Pull request by kmecpp**
Fixes #3196

### CODE REVIEW

**Comment by darbyjack** [comment]
I think this is a great idea! I have not tested it myself but I assume that it will properly handle when a user gives himself a kit vs someone else? As in, it will tie to the player receiving the kit?

Edit: Tested and confirmed my own question. It works!

**Review by darbyjack — DISMISSED** [review]
I have pulled this and tested it myself. It seems to work pretty nicely. I can confirm that it will properly display the user the kit was given to whether it was given by self-ran command or via console command.

Here's a little example I wrote using the event. It will listen for the event, and if it has armor, it will auto-equipt it!!

ezgif com-video-to-gif

File: Essentials/src/net/ess3/api/events/UserWarpEvent.java [hunk 1]
```diff
@@ -3,7 +3,6 @@
 import com.earth2me.essentials.Trade;
 import net.ess3.api.IUser;
 import org.bukkit.Bukkit;
-import org.bukkit.entity.Player;
```
Comment (mdcfe) [inline Essentials/src/net/ess3/api/events/UserWarpEvent.java]: This change is unrelated - could you revert it to clean up the diff?

File: Essentials/src/net/ess3/api/events/UserKitUseEvent.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+import org.bukkit.event.HandlerList;
+
+import com.earth2me.essentials.Kit;
+
+import net.ess3.api.IUser;
+
+/**
+ * Called when the player is given a kit
+ */
+public class UserKitUseEvent extends Event implements Cancellable {
```
Comment (mdcfe) [inline Essentials/src/net/ess3/api/events/UserKitUseEvent.java]: I'd prefer this was called `KitClaimEvent`, which is more descriptive about when the event is called.

**Review by mdcfe — APPROVED** [review]

**Review by darbyjack — APPROVED** [review]
